assets/js/stripe-payment-request.js

```diff
@@ -678,24 +690,22 @@ jQuery( function( $ ) {
 			}
 		},
 
-		blockPaymentRequestButton: function( prButton ) {
+		blockPaymentRequestButton: function( prButton, cssClassname ) {
 			// check if element isn't already blocked before calling block() to avoid blinking overlay issues
 			// blockUI.isBlocked is either undefined or 0 when element is not blocked
 			if ( $( '#wc-stripe-payment-request-button' ).data( 'blockUI.isBlocked' ) ) {
 				return;
 			}
 
-			$( '#wc-stripe-payment-request-button' ).block( { message: null } );
-			if ( wc_stripe_payment_request.isCustomPaymentRequestButton( prButton ) ) {
-				prButton.addClass( 'is-blocked' );
-			}
+			$( '#wc-stripe-payment-request-button' )
+				.addClass( cssClassname )
+				.block( { message: null } );
```

Comment:
Don't we have to include the custom PRB check + `prButton.addClass` call? Or has that been made redundant? The way this function is set up now makes it so that `prButton` is never used, and that doesn't seem right.

The same is true for the `unblockPaymentRequestButton` function.

Reply:
> Don't we have to include the custom PRB check + prButton.addClass call

Good Catch. I think we must add a classname wether PRB is "Custom" or not. Clearly, theme developers will want to customize the button not only when is custom. At least `storefront` does it. I proposed to add a classname to know exactly the state of the button [here](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1539#issuecomment-827909578). This way the theme will be able to stylize the PRB properly for each state of the button. Let me know what do you think about this.

Maybe I can also add the `is-blocked` classname for backwards compatibility but seems unnecessary since the [PR](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1197) that introduced it [never used it](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1197#discussion_r621580955) to customize the button. It looks like it was added to let theme developers to be able to customize the PRB but I doesn't seem it's documented or used at all.

> The way this function is set up now makes it so that prButton is never used, and that doesn't seem right.

Correct. Since `prButton` can be a `jQuery` object or a Stripe `Element` I'm adding the classname to the parent element `#wc-stripe-payment-request-button` since it wraps all button types, also to avoid adding the classname in different ways based on the object type. I'll remove `prButton` from the arguments. Let me know if this sounds good to you.

Reply:
I've removed the unused references of `prButton` let me know if this approach looks good to you

Reply:
> Let me know what do you think about [adding a class name to know the PRB state for theme developers]

I like the approach 🙂 

> Maybe I can also add the `is-blocked` classname for backwards compatibility

Based on your comment I don't think that's necessary ☺️ 

> I'm adding the classname to the parent element [...] since it wraps all button types

Nice!

>I've removed the unused references of `prButton` let me know if this approach looks good to you

That's what I was going to suggest. Thank you! ❤️